lib/openstax/aws/stack.rb

```diff
@@ -1,5 +1,58 @@
+require 'json'
+
 module OpenStax::Aws
   class Stack
+    class Event
+      def initialize(aws_stack_event)
+        @aws_stack_event = aws_stack_event
+      end
+            
+      def status
+        @aws_stack_event.data.resource_status
+      end
+            
+      def reason
+        @aws_stack_event.data.resource_status_reason
+      end
+            
+      def failure?
+        status.include?("_FAILED") || status.include?("ROLLBACK_COMPLETE")
+      end
+
+      def failure_with_reason?
```

Comment:
What's the difference between this and `failure?` above?  the "with_reason" naming difference isn't clear